### ISSUE TITLE
Checks not accurately launched with defined frequency

### ISSUE THREAD

**Issue by rkalicinski** [arachnys/cabot#368]
I have check defined with 5 minutes frequency for data points stored every 1 minute.

![3_cabot_check_definition](https://cloud.githubusercontent.com/assets/21097493/17772006/9b1dba80-6546-11e6-8261-25dbdd225533.png)

Unfortunately, above check is launched with not precise time span. Sometimes it is 5 minutes, another time could be 6 or even 7 minutes between each check.

![2_cabot_check_lists](https://cloud.githubusercontent.com/assets/21097493/17772064/f65c2f3a-6546-11e6-8243-0ccccbb3a857.png)

Finally it results with missing important data due to the slippage:

![1_cabot_check_result](https://cloud.githubusercontent.com/assets/21097493/17772140/6462a3ce-6547-11e6-9205-82ad54a08b9a.png)

In above graph metric with value 1 would be taken into account when there would be no slippage.

Is there any way to somehow workaround that issue?

**Comment by romanek-adam** [comment]
I was hit by this issue too. We've been struggling with this Cabot issue for quite some time and we haven't found any workaround yet.

Is there any hope this could get fixed soon? It's a critical issue as it may result in false positives and false negatives.

**Comment by dbuxton** [comment]
I agree that this isn't great.

I suggest using #366 and modifying it so that it takes `max(frequency, time_since_last_check_started)` as the time period to check. That way the check will look at all the data since the last check ran.

**Comment by dbuxton** [comment]
OK I have updated #366 to do this. @romanek-adam @rkalicinski could you try deploying that branch to see if it behaves as expected?

@JeanFred FYI

**Comment by romanek-adam** [comment]
Sure, I will test it. Thanks!

**Comment by rkalicinski** [comment]
I will also try it out. Many thanks!

**Comment by rkalicinski** [comment]
I deployed Cabot from `read-graphite-timestamps` branch. It works way better now, so thanks @dbuxton for your help. Nevertheless, there is still an issue with fetching too few metrics.
I have a check with 6 minutes frequency and it fires up with 6, 7, 8 minutes interval. In image below I marked the problematic situation where there was a 7 minutes gap between two successive checks.
![cabot_1](https://cloud.githubusercontent.com/assets/21097493/18195443/b811ef66-70eb-11e6-9b3a-0e1a570eff5d.png)
In above case, latter check only fetched 6 metrics (1 minute interval) and I suppose it should base on 7 metrics instead. 
![cabot_2](https://cloud.githubusercontent.com/assets/21097493/18195654/f742d104-70ec-11e6-9462-010dfd5d717d.png)
Maybe it would be necessary to add some safety buffer in such case, in order not to miss some data that may come between successive checks?
